python/dgl/nn/pytorch/conv/cugraph_gatconv.py

```diff
@@ -170,25 +172,17 @@ def forward(self, g, feat, max_in_degree=None):
                 max_in_degree = g.in_degrees().max().item()
 
             if max_in_degree < self.MAX_IN_DEGREE_MFG:
-                _graph = make_mfg_csr(
-                    g.dstnodes(),
+                _graph = SampledCSC(
                     offsets,
                     indices,
                     max_in_degree,
                     g.num_src_nodes(),
                 )
             else:
-                offsets_fg = torch.empty(
-                    g.num_src_nodes() + 1,
-                    dtype=offsets.dtype,
-                    device=offsets.device,
-                )
-                offsets_fg[: offsets.numel()] = offsets
-                offsets_fg[offsets.numel() :] = offsets[-1]
-
-                _graph = make_fg_csr(offsets_fg, indices)
+                offsets_fg = self.pad_offsets(offsets, g.num_src_nodes() + 1)
```

Comment:
Could you elaborate more about the motivations of zero-padding?

Reply:
We provide two variants of aggregation functions for a specific model. One intended for sampled-graph use cases, one intended for full-graph ones. The former is in general more performant, however, it only works when the sample size (or the max of in-degrees) is small (<200), due to the limit of GPU shared memory. When dealing with graph that have a large max in-degree, we need to fall back to the full-graph option. This requires us to convert a DGL block to a full graph, a square adjacency matrix. With the csc-representation, this is equivalent to pad zero-in-degree nodes to the end of the offsets array (also called indptr or colptr).

Reply:
Thanks. Could you include this description in the file as comments?

Reply:
Yes, I expanded the docstring of `CuGraphBaseConv.pad_offsets()`.